lib/readline.js

```diff
@@ -47,9 +47,9 @@ function Interface(input, output, completer, terminal) {
   }
   historySize = historySize || kHistorySize;
 
-  completer = completer || function() { return []; };
+  completer = completer || undefined;
 
-  if (typeof completer !== 'function') {
+  if (! typeof completer in ['function', 'undefined']) {
```

Comment:
`(typeof completer !== 'function || typeof completer !== 'undefined')` will probably be inlined better by v8 / JS vm's.